models/idea/index.js

```diff
@@ -308,6 +308,52 @@ class Idea extends Model {
     const cursor = await this.db.query(query, params);
     return await cursor.all();
   }
+
+
+  /**
+   * Read trending ideas
+   * @param {integer} offset - pagination offset
+   * @param {integer} limit - pagination limit
+   * @returns {Promise<Idea[]>} - list of found ideas
+   */
+  static async findTrending({ offset, limit }) {
+    const now = + new Date();
+    const oneWeek = 604800000; // 1000 * 60 * 60 * 24 * 7
+    const threeWeeks = 1814400000; // 1000 * 60 * 60 * 24 * 21
+    const threeMonths = 7776000000; // 1000 * 60 * 60 * 24 * 90
+    const weekAgo = now - oneWeek;
+    const threeWeeksAgo = now - threeWeeks;
+    const threeMonthsAgo = now - threeMonths;
+
+    // for each idea we are counting 'rate'
+    // rate is the sum of votes/day in the last three months
+    // votes/day from last week are taken with wage 3
+    // votes/day from two weeks before last week are taken with wage 2
+    // votes/day from the rest of days are taken with wage 1
+    const query = `
+      FOR idea IN ideas
```

Comment:
I got idea how to simplify this:

`FOR vote IN votes FILTER IS_SAME_COLLECTION('ideas', vote._to) COLLECT ideaId = votes._to AGGREGATE ... LET idea = DOCUMENT(ideaId) ...`

We get rid of one FOR, otherwise they do the same. And it's a one-liner, too. :stuck_out_tongue: 
I don't require this change. :)